src/ITensors.jl

```diff
@@ -182,3 +182,13 @@ end
 # TODO: make this definition AbstractMPS
 # Handle orthogonality center correctly
 Base.getindex(ψ::MPS, r::UnitRange{Int}) = MPS([ψ[n] for n in r])
+
+#TODO Remove if everything is working nicely
+#Was still crashing on my laptop after updating ITensors
+Base.fill!(::NDTensors.NoData, ::Any) = NDTensors.NoData()
+
+function ITensors.NDTensors.contraction_output(
+  A::NDTensors.EmptyTensor, B::NDTensors.DiagBlockSparseTensor, label
+)
+  return NDTensors.EmptyTensor(eltype(B), label)
```

Comment:
I think `promote_type(eltype(A), eltype(B))` instead of `eltype(B)` would be more general and robust.

I would argue for moving this to NDTensors.jl, but we are working on removing the `EmptStorage` type in favor of a much simpler design in https://github.com/ITensor/ITensors.jl/pull/1161.